src/include/platform/ConnectivityManager.h

```diff
@@ -190,6 +194,9 @@ class ConnectivityManager
     void MaintainOnDemandWiFiAP();
     System::Clock::Timeout GetWiFiAPIdleTimeout();
     void SetWiFiAPIdleTimeout(System::Clock::Timeout val);
+    CHIP_ERROR GetNetworkSSID(char * ssid);
+    CHIP_ERROR GetNetworkPassword(char * credentials);
```

Comment:
Those signatures do not look ok from a "no buffer overruns" point of view.

Reply:
Are you referring that I should add a check for null on the ` char * ` in the `GetNetworkSSID` and `GetNetworkPassword` definitions?

Reply:
@bzbarsky-apple can you give an example on how to do this, please?

Reply:
You should be using MutableCharSpan out params for handing out strings, ideally.

Reply:
Hello. I agree that MutableCharSpan is a safer option to handle strings, but in this case, it will add extra dependencies to the ConnectivityManager.h file by introducing the MutableCharSpan header file. There are other functions in the same file handling strings in a similar way, like CHIP_ERROR GetBLEDeviceName(char * buf, size_t bufSize); and CHIP_ERROR SetBLEDeviceName(const char * deviceName);

Reply:
Yes, there are existing bad APIs that came with the code import.  We should not add more bad, memory-unsafe, APIs.

Reply:
I agree, I updated the implementation. Thanks for feedback.